var/spack/repos/builtin/packages/py-graphql-relay/package.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyGraphqlRelay(PythonPackage):
+    """Relay library for graphql-core."""
+
+    homepage = "https://github.com/graphql-python/graphql-relay-py"
+    pypi = "graphql-relay/graphql-relay-2.0.1.tar.gz"
+
+    maintainers("LydDeb")
+
+    version("2.0.1", sha256="870b6b5304123a38a0b215a79eace021acce5a466bf40cd39fa18cb8528afabb")
+
+    depends_on("py-setuptools", type="build")
+    depends_on("py-graphql-core@2.2:2", type=("build", "run"), when="@2")
+    depends_on("py-six@1.12:", type=("build", "run"), when="@2")
+    depends_on("py-promise@2.2:2", type=("build", "run"), when="@2")
+    # spack test failed with message: ModuleNotFoundError: No module named rx
+    depends_on("py-rx", type=("build", "run"))
```

Comment:
Tell me more about this error? I grepped the entire source directory but "rx" does not exist.

Reply:
My bad, I did not notice that was a *ModuleNotFoundError* from py-graphql-**core** and not py-graphql-**relay**:
```
==> [2023-09-26-10:09:44.115284] test: test_imports_graphql: checking import of graphql
==> [2023-09-26-10:09:44.115514] '/opt/spack/linux-rocky8-zen/gcc-8.5.0/python-3.8.10-lesqjyytmmorjygjfy5aidyqlkh4rbgg/bin/python3.8' '-c' 'import graphql'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/opt/spack/linux-rocky8-zen/gcc-8.5.0/py-graphql-core-2.3.2-bgivakkwpd7st77auxza4yosiflf4whm/lib/python3.8/site-packages/graphql/__init__.py", line 27, in <module>
    from .graphql import graphql
  File "/opt/spack/linux-rocky8-zen/gcc-8.5.0/py-graphql-core-2.3.2-bgivakkwpd7st77auxza4yosiflf4whm/lib/python3.8/site-packages/graphql/graphql.py", line 1, in <module>
    from .execution import ExecutionResult
  File "/opt/spack/linux-rocky8-zen/gcc-8.5.0/py-graphql-core-2.3.2-bgivakkwpd7st77auxza4yosiflf4whm/lib/python3.8/site-packages/graphql/execution/__init__.py", line 21, in <module>
    from .executor import execute, subscribe
  File "/opt/spack/linux-rocky8-zen/gcc-8.5.0/py-graphql-core-2.3.2-bgivakkwpd7st77auxza4yosiflf4whm/lib/python3.8/site-packages/graphql/execution/executor.py", line 11, in <module>
    from rx import Observable
ModuleNotFoundError: No module named 'rx'
```
I will remove this dependence in the recipe of py-graphql-relay and add it in the recipe of py-graphql-core for version 2.3.2